docs/core/diagnostics/dotnet-counters.md

```diff
@@ -15,6 +15,12 @@ To install the latest release version of the `dotnet-counters` [NuGet package](h
 dotnet tool install --global dotnet-counters
 ```
 
+Starting from .NET 5, it is now possible to directly download a one-file executable for your operating system:
```

Comment:
I don't think we need to mention `Starting from .NET 5`.  Here or in each document.

```suggestion
It is also possible to directly download a single-file executable:
```

I am not sure of the placement here.  It is possible the links could be below. It is also possible we might want them all on a single reference page.

We will need to update for Apple Silicon later this year too.

Reply:
I hadn't added the documentation because adding it per tool has two issues:
- It doesn't call out that you need a 3.1 runtime or any other caveats like access to TEMP in the environment (I already hit this with a few customers). 
- It is not a preferred mechanism. There's no update, cleanup, or scope story around this. 

I was thinking of just adding a note to each tool saying: `In case you need to use this tool in an environment where the SDK is not available, please refer to ...`. I was talking about this with @sywhang, but he caveated the one thing I don't like of my approach: now users need to check two pages. Any feedback @dotnet/dotnet-diag @chrisnas?

Reply:
Thought about this for a while last night, and what I think we can do is to update each tool's page to include a link to the main reference page that has the caveats mentioned. Since single-file tools are not the preferred mechanisms for the reasons Juan mentioned above, it's probably better to just mention that it is possible on each tools' page, and user can go through one more step to download this less-preferred mechanism if they need it.

Reply:
Are you sure that .NET 5+ versions of the tools will work with .NET Core 3 apps?

Reply:
Yes. Tools are written in a way that will fall back if it tries to use a feature that is not compatible, but for all features that are compatible with 3.1, the tools will/should work.

Reply:
> I don't think we need to mention `Starting from .NET 5`. Here or in each document.
> 
Are you sure that .NET 5+ versions of the tools will work with .NET Core 3/3.1 apps?

Reply:
> Are you sure that .NET 5+ versions of the tools will work with .NET Core 3/3.1 apps?

Yes. Tools are written in a way that will fall back if it tries to use a feature that is not compatible, but for all features that are compatible with 3.1, the tools will/should work.

Reply:
> * It doesn't call out that you need a 3.1 runtime or any other caveats like access to TEMP in the environment (I already hit this with a few customers).

That's why I added the ".NET 5 bla bla" at the beginning. But a better wording could be found such as "Starting with .NET Core 3.1 <or any relevant version>, bla bla"

> * It is not a preferred mechanism. There's no update, cleanup, or scope story around this.
> I was thinking of just adding a note to each tool saying: `In case you need to use this tool in an environment where the SDK is not available, please refer to ...`. I was talking about this with @sywhang, but he caveated the one thing I don't like of my approach: now users need to check two pages. Any feedback @dotnet/dotnet-diag @chrisnas?

This will definitively be the ONLY version used in production: nobody will install the SDK in production JUST to download a tool. And I'm not talking about containers that must be as small as possible. 
You are right about the known limitations: they must be stated clearly.
I was thinking about adding a "Download" section instead of adding the text under the "Install" section
The download link should be in the tool page, not in another side page.
If you look at what is done for SysInternals, each tools has its download link and there is another page for the Suite where all tools could be downloaded at once. That could be a solution to keep the links for each tool + a common page with the limitations + all the links

Reply:
> > Are you sure that .NET 5+ versions of the tools will work with .NET Core 3/3.1 apps?
> 
> Yes. Tools are written in a way that will fall back if it tries to use a feature that is not compatible, but for all features that are compatible with 3.1, the tools will/should work.

The text mentions .NET SDK 3.0; nothing about 3.1 runtime: this is why I added the mention to .NET 5

Reply:
The regular tools are `netcoreapp2.1` with a roll-forward policy. The self-contained apps are 3.1 - as the first version of single file was added there. The tools are agnostic of the runtime the target app is running, as long as there's a proper pipe it will work (so the target app needs 3.1+, and potentially 5.0 for some features like events at startup and features that rely on reverse communication mechanisms)

Reply:
I don't think we should frame the download options as being more or less 'preferred'. We want customers to use the best option for their situation. For many of them, the best option is the single-file download.



In terms of the presentation here, how does this sound?
 
## Install dotnet-counters

**Option 1** - Install as a [dotnet global tool](../tools/dotnet-tool-install.md):
```dotnetcli
dotnet tool install --global dotnet-counters
```

**Option 2** - Direct download:
- Win ([x86](https://aka.ms/dotnet-counters/win-x86) | [x64](https://aka.ms/dotnet-counters/win-x64) | [arm](https://aka.ms/dotnet-counters/win-arm) | [arm-x64](https://aka.ms/dotnet-counters/win-arm64))
- macOS ([x64](https://aka.ms/dotnet-counters/osx-x64))
- Linux ([x64](https://aka.ms/dotnet-counters/linux-x64) | [arm](https://aka.ms/dotnet-counters/linux-arm) | [arm64](https://aka.ms/dotnet-counters/linux-arm64) | [musl-x64](https://aka.ms/dotnet-counters/linux-musl-x64) | [musl-arm64](https://aka.ms/dotnet-counters/linux-musl-arm64))

For guidance on installation options and system requirements see \<some_link_to_a_page_with_details>